### PR TITLE
[wip] Add support to workers

### PR DESCRIPTION
**Pull request by yourwebmaker**
This pull request aims to add Workers support to dynamic configuration.

It is also based on existent configuration found in Puphpet: https://puphpet.com/#beanstalkd

Initially, we'll support only Beanstalkd and RabbitMQ servers.

- [x] **RabbitMQ**: Installation based on https://www.digitalocean.com/community/tutorials/how-to-install-and-manage-rabbitmq + rabbitmq_* modules.
- [x] **Beanstalked**: No idea :bulb: but I guess it's so easy as sunday morning.

### CODE REVIEW

File: tests/src/Phansible/Roles/RoleTest.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        $role = $this->getMockBuilder('\Phansible\BaseRole')
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        $role->expects($this->once())
+            ->method('getInitialValues')
+            ->will($this->returnValue([]));
+
+        $this->assertInternalType('array', $role->getInitialValues());
```
Comment (naxhh) [inline tests/src/Phansible/Roles/RoleTest.php]: This test makes no sense.

You told the baserole to return empty array and then you assert that returns an array.
What you win checking this?

File: tests/src/Phansible/Roles/RedisTest.php [hunk 2]
```diff
@@ -1,25 +0,0 @@
-<?php
```
Comment (naxhh) [inline tests/src/Phansible/Roles/RedisTest.php]: Why this was removed?

File: app/config/phansible/syspackages.yml [hunk 3]
```diff
@@ -7,3 +7,4 @@ syspackages:
   - imagemagick
   - nodejs
   - npm
+
```
Comment (naxhh) [inline app/config/phansible/syspackages.yml]: Please don't add changes that are not part of the PR
Or changes that don't add anything.

File: tests/src/Phansible/Roles/RoleTest.php (hunk 1 above)
Comment (yourwebmaker) [inline tests/src/Phansible/Roles/RoleTest.php]: Sorry... my bad. I was trying to avoid test duplication through abstracting all BaseRole implementations in only one test. But it seems I need to create one test for each implementation.

Thanks.

File: app/config/phansible/syspackages.yml (hunk 3 above)
Comment (yourwebmaker) [inline app/config/phansible/syspackages.yml]: got it.

**Comment by yourwebmaker** [comment]
@erikaheidi @naxhh  ... could you check this one?

File: src/Phansible/Resources/ansible/roles/beanstalkd/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+galaxy_info:
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/beanstalkd/meta/main.yml]: I'm a bit rusty with ansible.
This will require you to have galaxy installed?

File: src/Phansible/Resources/ansible/roles/beanstalkd/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  author: "Jasper N. Brouwer, Ramon de la Fuente"
+  description: Install and start beanstalkd
+  company: Future500
+  license: LGPL
+  min_ansible_version: 1.4
+  platforms:
+  - name: Debian
+    versions:
+      - wheezy
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/beanstalkd/meta/main.yml]: Keep in mind we have more versions. not just this one.

File: src/Phansible/Resources/ansible/roles/beanstalkd/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+---
+- name: Discover if there is already a version of beanstalkd installed.
+  shell: which beanstalkd > /dev/null && beanstalkd -v | cut -d' ' -f2
+  changed_when: False
+  failed_when: False
+  register: installed_beanstalkd_version
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/beanstalkd/tasks/main.yml]: What this holds when the first command fails? And how that affects the rest of the playbook?

File: src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: add the verification key for the package
+  sudo: yes
+  shell: curl http://www.rabbitmq.com/rabbitmq-signing-key-public.asc | sudo apt-key add -
+
+- name: update repositories
+  apt: update_cache=yes
+
+- name: install rabbitmq
+  apt: pkg=rabbitmq-server state=installed force=yes
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml]: Why you need to force this?

File: src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: enable rabbitmq plugins
+  rabbitmq_plugin: names={{ item }} state=enabled
+  with_items: rabbitmq.plugins
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml]: Can you paste here any doc that explains why this plugins are needed?

File: src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  when: rabbitmq.plugins is defined
+
+- name: add user
+  rabbitmq_user: user={{ rabbitmq.user }} password={{ rabbitmq.password }} node=rabbit@{{ vagrant_local.vm.hostname }} tags=administrator,{{ rabbitmq.user }} vhost=/ configure_priv=.* write_priv=.* read_priv=.* state=present
+
+- name: restart rabbitmq
+  service: name=rabbitmq-server state=reloaded
```
Comment (naxhh) [inline src/Phansible/Resources/ansible/roles/rabbitmq/tasks/main.yml]: This will restart rabbit every time even if changes are not performed.
Shouldn't be better to restart it only when required?

**Comment by naxhh** [comment]
Nice job.

Can you make a list of the end2end tests you have performed?
Boxes where you tried it, if you needed to test any weird convination (like benstalk + php because install a specific plugin, etc...)

Testing this things takes me a lot of time and knowing what you already tested would be nice.
